prov/sockets/src/sock_ep.c

```diff
@@ -1670,18 +1672,19 @@ int sock_alloc_endpoint(struct fid_domain *domain, struct fi_info *info,
 	if (sock_conn_map_init(sock_ep, sock_cm_def_map_sz)) {
 		SOCK_LOG_ERROR("failed to init connection map: %s\n", strerror(errno));
 		ret = -FI_EINVAL;
-		goto err;
+		goto err2;
 	}
 
 	atomic_inc(&sock_dom->ref);
 	return 0;
 
-err:
+err2:
 	if (sock_ep->attr->src_addr)
 		free(sock_ep->attr->src_addr);
 	if (sock_ep->attr->dest_addr)
 		free(sock_ep->attr->dest_addr);
```

Comment:
Yes, but I think we should check if sock_ep->attr is not null instead of sock_ep->attr->src_addr. Otherwise we could have a null pointer dereference.